react/features/contact-list/components/ContactListPanel.web.js

```diff
@@ -0,0 +1,182 @@
+
+const { PropTypes } = React;
+
+declare var interfaceConfig: Object;
+
+/**
+ * React component for showing a list of current conference participants, the
+ * current conference lock state, and a button to open the invite dialog.
+ *
+ * @extends Component
+ */
+class ContactListPanel extends Component {
+    /**
+     * Default values for {@code ContactListPanel} component's properties.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * Whether or not the conference is currently locked with a password.
+         */
+        _locked: PropTypes.bool,
+
+        /**
+         * The participants to show in the contact list.
+         */
+        _participants: PropTypes.array,
+
+        /**
+         * Invoked to open an invite dialog.
+         */
+        dispatch: PropTypes.func,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code ContactListPanel} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind event handler so it is only bound once for every instance.
+        this._onOpenInviteDialog = this._onOpenInviteDialog.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { _locked, _participants, t } = this.props;
+
+        return (
+            <div className = 'contact-list-panel'>
+                <div className = 'title'>
+                    { t('contactlist', { pcount: _participants.length }) }
+                </div>
+                <div className = 'sideToolbarBlock first'>
+                    <Button
+                        appearance = 'primary'
+                        className = 'contact-list-panel-invite-button'
+                        id = 'addParticipantsBtn'
+                        onClick = { this._onOpenInviteDialog }
+                        type = 'button'>
+                        { t('addParticipants') }
+                    </Button>
+                    <div>
+                        { _locked
+                            ? this._renderLockedMessage()
+                            : this._renderUnlockedMessage() }
+                    </div>
+                </div>
+                <ul id = 'contacts'>
+                    { this._renderContacts() }
+                </ul>
+            </div>
+        );
+    }
+
+    /**
+     * Dispatches an action to open an invite dialog.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onOpenInviteDialog() {
+        this.props.dispatch(openInviteDialog());
+    }
+
+    /**
+     * Renders React Elements for displaying information about each participant
+     * in the contact list.
+     *
+     * @private
+     * @returns {ReactElement[]}
+     */
+    _renderContacts() {
```

Comment:
There's some inconsistency currently in the naming that's coming from the legacy code before this PR. We "renderContacts", but get them from a participants' list, for example. To add to the confusion we renamed them to "members" in the translation :) Not sure if that's the right time to fix this, but can you add a FIXME comment about this somewhere in this file, please?

Reply:
Contacts vs participants might be a tricky one to fix, especially because the button name is called contacts.